File: 20180713/Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res = [[1 for _ in range(level + 1)] for level in range(numRows)]
        for i in range(1, numRows):
            for j in range(1, i):
                res[i][j] = res[i-1][j-1] + res[i-1][j]
        return res
                

File: 20180713/Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        for(int level = 0; level < numRows; level ++){
            List<Integer> tmp = new ArrayList<Integer>();
            for(int i = 0; i < level + 1; i ++){
                tmp.add(1);
            }
            res.add(tmp);
        }
        for(int level = 1; level < numRows; level ++){
            for(int i = 1; i < level; i ++){
                res.get(level).set(i,  res.get(level - 1).get(i - 1) + res.get(level - 1).get(i));
            }
        }
        return res;
    }
}
